python/GafferUITest/TestCase.py

```diff
@@ -83,6 +83,29 @@ def assertExampleFilesExist( self ) :
 			expanded = os.path.expandvars( e['filePath'] )
 			self.assertTrue( os.path.exists( expanded ), "%s does not exist" % expanded )
 
+	def assertExampleFileDontReferenceUnstablePaths( self ) :
+
+		forbidden = (
+			"${script:name}",
+			"/home/"
+		)
+
+		safePlugNames = (
+			"title",
+			"description"
+		)
+
+		examples = GafferUI.Examples.registeredExamples()
+		for e in examples.values():
+			path = os.path.expandvars( e['filePath'] )
+			with open( path, 'r' ) as example :
+				for line in example :
+					# If the line contains a set for one of our safe plugs, don't check
+					if filter( lambda plug : '["%s"].setValue(' % plug in line, safePlugNames ) :
```

Comment:
https://docs.quantifiedcode.com/python-anti-patterns/readability/using_map_or_filter_where_list_comprehension_is_possible.html
https://google.github.io/styleguide/pyguide.html#215-deprecated-language-features

Reply:
Haha, I had originally written it as a list comprehension, but felt it was harder to understand the intent 😆 Fixed in https://github.com/GafferHQ/gaffer/pull/3309/commits/549381d8a6c0d876247ecc5e2dbe18014584ba18

Reply:
It might just be me, but my mental parser really struggles with `filter` and `map`. I find the connection between `safePlugNames` at the end and `lambda plug` at the start really unobvious, and the extra lambda boilerplate just gets in the way anyway. I prefer just to avoid them entirely.

While I'm splitting hairs, perhaps this would be most expressive of intent?

```
if any( '["%s"].setValue(' % plug in line for plug in safePlugNames ) :
```

Reply:
Thats better than all of them! https://github.com/GafferHQ/gaffer/commit/74bc768b0e868262453c14da6c79511d0ae5ce6c